sourcerer-esjc/src/main/java/org/elder/sourcerer2/esjc/EventStoreEsjcEventRepository.kt

```diff
@@ -0,0 +1,553 @@
+                        || ex.cause is PersistentSubscriptionDeletedException
+                        || ex.cause is StreamDeletedException) {
+                    throw PermanentEventReadException(ex.cause)
+                } else if (ex.cause is CannotEstablishConnectionException
+                        || ex.cause is ClusterException
+                        || ex.cause is ConnectionClosedException
+                        || ex.cause is OperationTimedOutException
+                        || ex.cause is MaximumSubscribersReachedException
+                        || ex.cause is RetriesLimitReachedException
+                        || ex.cause is ServerErrorException) {
+                    throw RetriableEventReadException(ex.cause)
+                } else {
+                    logger.warn("Unrecognized event store exception reading events", ex.cause)
+                    throw RetriableEventReadException(ex.cause)
+                }
+            } else if (ex.cause is RuntimeException) {
+                logger.warn("Unrecognized runtime exception reading events", ex.cause)
+                throw RetriableEventReadException(ex.cause)
+            } else {
+                logger.warn("Unrecognized exception reading events", ex.cause)
+                throw RetriableEventReadException(
+                        "Internal error reading events",
+                        ex.cause)
+            }
+        } catch (ex: TimeoutException) {
+            throw RetriableEventReadException("Timeout reading events", ex.cause)
+        }
+
+    }
+
+    private fun <U> completeWriteFuture(
+            future: CompletableFuture<U>,
+            expectedVersion: ExpectedVersion
+    ): U {
+        try {
+            return future.get(timeoutMillis, TimeUnit.MILLISECONDS)
+        } catch (ex: InterruptedException) {
+            Thread.currentThread().interrupt()
+            throw RetriableEventWriteException("Internal error writing event", ex)
+        } catch (ex: ExecutionException) {
+            if (ex.cause is EventStoreException) {
+                if (ex.cause is WrongExpectedVersionException) {
+                    logger.warn("Unexpected version when completing EventStore write", ex.cause)
+                    throw UnexpectedVersionException(
+                            ex.cause, null,
+                            expectedVersion)
+                } else if (ex.cause is AccessDeniedException
+                        || ex.cause is CommandNotExpectedException
+                        || ex.cause is InvalidTransactionException
+                        || ex.cause is NoResultException
+                        || ex.cause is NotAuthenticatedException
+                        || ex.cause is PersistentSubscriptionDeletedException
+                        || ex.cause is StreamDeletedException) {
+                    throw PermanentEventWriteException(ex.cause)
+                } else if (ex.cause is CannotEstablishConnectionException
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Complex conditions should be simplified and extracted into well-named methods if necessary. SIZE 7 is greater than the threshold 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.elder.org/coding_rules#rule_key=detekt-kotlin%3AComplexCondition)